client/core/trade.go

```diff
@@ -574,19 +574,17 @@ func (t *trackedTrade) processCancelMatch(msgMatch *msgjson.Match) error {
 //
 // This method accesses match fields and MUST be called with the trackedTrade
 // mutex lock held for reads.
-func (t *trackedTrade) counterPartyConfirms(match *matchTracker) (have, needed uint32, changed bool) {
+func (t *trackedTrade) counterPartyConfirms(ctx context.Context, match *matchTracker) (have, needed uint32, changed, spent bool) {
 	// Counter-party's swap is the "to" asset.
 	needed = t.wallets.toAsset.SwapConf
 
 	// Check the confirmations on the counter-party's swap.
 	coin := match.counterSwap.Coin()
-	var err error
-	ctx, cancel := context.WithTimeout(context.Background(), confCheckTimeout)
-	defer cancel()
-	have, err = coin.Confirmations(ctx)
+
+	have, spent, err := t.wallets.toWallet.Confirmations(ctx, coin.ID())
 	if err != nil {
-		t.dc.log.Errorf("Failed to get confirmations of the counter-party's swap %s (%s) for match %v, order %v",
-			coin, t.wallets.toAsset.Symbol, match.id, t.UID())
+		t.dc.log.Errorf("Failed to get confirmations of the counter-party's swap %s (%s) for match %v, order %v: %v",
+			coin, t.wallets.toAsset.Symbol, match.id, t.UID(), err)
 		have = 0 // should already be
```

Comment:
Considering that `counterPartyConfirms` returns `spent=true` when `err != nil`, this is going to tell the caller that the txn (contract) is spent when really it just wasn't found.  Could be the case due to network latency or whatever else.